pages/ownersPage.ts

```diff
@@ -13,22 +13,76 @@ export class OwnersPage extends HelperBase{
         return this.page.getByRole('row', {name: owner})
     }
 
-    async selectOwner(owner: string){
-        await this.page.getByText(owner).click()
+    async clickOwnerNameFor(ownerName: string){
+        await this.page.getByText(ownerName).click()
     }
 
-    async validatePetNameAndCityForOwner(petName: string, petOwner: string, city: string){
-        const row = this.page.getByRole('row', {name: 'Jeff Black'})
+    async validatePetAndCityOfOwner(owner: string, petName: string, city: string){
+        const row = this.page.getByRole('row', {name: owner})
     
         // Validate the owner is from Monona
-        await expect(row.locator('td').nth(2)).toHaveText('Monona')
+        await expect(row.locator('td').nth(2)).toHaveText(city)
     
         // Validate the owner has a pet named Lucky
-        await expect(row.locator('td').nth(4)).toHaveText('Lucky')
+        await expect(row.locator('td').nth(4)).toHaveText(petName)
         
     }
 
-    async validatePetNumberPerCity(city: string){
-        await expect(this.page.getByRole('row', {name: city})).toHaveCount(4)
+    async getOwnerCountForCity(city: string){
+        return this.page.getByRole('row', {name: city}).all()
     }
+
+    async validateOwnerCountForCity(city: string, count: number){
+        await expect(this.page.getByRole('row', {name: city})).toHaveCount(count)
+    }
+
+    async validatePetCountForCity(city: string, petList: string[]){
+         await this.page.waitForSelector('table')
+
+         const allRows = await this.getOwnerCountForCity(city)
+        
+        let actualPetsList: string[] = []
+    
+        // Iterate through the rows to extract the pet names
+        for(const row of allRows){
+            // Add the pet name to the comparing list of names, with each name trimmed of excess white space
+            actualPetsList.push( (await row.getByRole('cell').nth(4).textContent())?.trim()! )
+        }
+    
+        // Comparing both sorted lists for equality
+        expect (actualPetsList.sort()).toEqual(petList.sort())
+
+    }
+
+    async searchForOwnerByLastName(ownerLastName: string){
```

Comment:
This name does not reflect the result. Search owner and do what? :)

Reply:
Hmm, ok, I am also checking for the existence of owners, so I have come up with `searchForOwnerByLastNameAndValidateOwnerExistsFor(ownerLastName: string)`